File: blackhole/security.kt
import io.ktor.application.*
import io.ktor.response.*
import io.ktor.routing.*
import io.ktor.http.*
import io.ktor.server.engine.*
import io.ktor.server.netty.*

fun main() {
    embeddedServer(Netty, port = 8080) {
        routing {
            get("/") {
                call.response.headers.append(HttpHeaders.ContentSecurityPolicy, "default-src 'self'")
                call.response.headers.append(HttpHeaders.XFrameOptions, "DENY")
                call.response.headers.append(HttpHeaders.XContentTypeOptions, "nosniff")
                call.response.headers.append(HttpHeaders.StrictTransportSecurity, "max-age=63072000; includeSubDomains; preload")
                call.respondText("<h1>Secure Kotlin Web Page</h1>", ContentType.Text.Html)
            }
        }
    }.start(wait = true)
}

File: blackhole/security.go
package main

import (
	"fmt"
	"net/http"
)

func secureHeaders(next http.Handler) http.Handler {
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Content-Security-Policy", "default-src 'self'")
		w.Header().Set("X-Frame-Options", "DENY")
		w.Header().Set("X-Content-Type-Options", "nosniff")
		w.Header().Set("Strict-Transport-Security", "max-age=63072000; includeSubDomains; preload")
		next.ServeHTTP(w, r)
	})
}

func handler(w http.ResponseWriter, r *http.Request) {
	fmt.Fprint(w, "<h1>Secure Go Web Page</h1>")
}

func main() {
	mux := http.NewServeMux()
	mux.Handle("/", secureHeaders(http.HandlerFunc(handler)))
	http.ListenAndServe(":8080", mux)
}